pkg/oceancd/builders/rollout.go

```diff
@@ -0,0 +1,94 @@
+package builders
+
+import (
+	"spot-oceancd-cli/pkg/oceancd"
+	"spot-oceancd-cli/pkg/oceancd/model/rollout"
+	"spot-oceancd-cli/pkg/oceancd/repositories"
+	"sync"
+)
+
+type DetailedRolloutBuilder struct {
+	detailedRollout   *rollout.DetailedRollout
+	rolloutRepository *repositories.RolloutRepository
+	errors            chan error
+	wg                *sync.WaitGroup
+	withStrategy      bool
+}
+
+func NewDetailedRolloutBuilder(rolloutRepository *repositories.RolloutRepository) *DetailedRolloutBuilder {
+	return &DetailedRolloutBuilder{
+		rolloutRepository: rolloutRepository,
+	}
+}
+
+func (b *DetailedRolloutBuilder) Build(rolloutId string) (*rollout.DetailedRollout, error) {
+	b.detailedRollout = &rollout.DetailedRollout{}
+	b.errors = make(chan error)
+
+	b.wg = &sync.WaitGroup{}
+
+	if b.withStrategy {
+		//add setStrategy()
+		b.wg.Add(1)
+		go b.setStrategy(rolloutId)
+	}
+
+	//add setRollout()
```

Comment:
mention that this group waits to go routine of `setStrategy`
It's a comment don't be shy with words :)

Reply:
do the same to the other comments